mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -1,409 +1,74 @@
-
-    early_stop_callback = None
-    for callback in self.callbacks:
-        if isinstance(callback, pl.callbacks.early_stopping.EarlyStopping):
-            early_stop_callback = callback
-            _log_early_stop_params(early_stop_callback, client, run_id)
-
-    if not any(isinstance(callbacks, __MLflowPLCallback) for callbacks in self.callbacks):
-        self.callbacks += [
-            __MLflowPLCallback(
-                client, metrics_logger, run_id, log_models, log_every_n_epoch, log_every_n_step
-            )
-        ]
-
-    client.flush(synchronous=False)
 
+def patched_add_event(original, self, event, *args, mlflow_log_every_n_step, **kwargs):
+    run = mlflow.active_run()
+    if (
+        not ENABLED
+        and run is not None
+        and event.WhichOneof("what") == "summary"
+        and mlflow_log_every_n_step
+    ):
+        summary = event.summary
+        global_step = args[0] if len(args) > 0 else kwargs.get("global_step", None)
+        global_step = global_step or 0
+        for v in summary.value:
+            if v.HasField("simple_value"):
+                if global_step % mlflow_log_every_n_step == 0:
+                    _add_to_queue(
+                        key=v.tag,
+                        value=v.simple_value,
+                        step=global_step,
+                        time=int((event.wall_time or time.time()) * 1000),
+                        run_id=run.info.run_id,
+                    )
+
+    return original(self, event, *args, **kwargs)
+
+
+def patched_add_summary(original, self, *args, **kwargs):
     result = original(self, *args, **kwargs)
-
-    if early_stop_callback is not None:
-        _log_early_stop_metrics(early_stop_callback, client, run_id)
-
-    if Version(pl.__version__) < Version("1.4.0"):
-        # pylint: disable-next=unexpected-keyword-arg
-        summary = str(ModelSummary(self.model, mode="full"))
-    else:
-        summary = str(ModelSummary(self.model, max_depth=-1))
-
-    tempdir = tempfile.mkdtemp()
-    try:
-        summary_file = os.path.join(tempdir, "model_summary.txt")
-        with open(summary_file, "w") as f:
-            f.write(summary)
-
-        mlflow.log_artifact(local_path=summary_file)
-    finally:
-        shutil.rmtree(tempdir)
-
-    if log_models:
-        registered_model_name = get_autologging_config(
-            mlflow.pytorch.FLAVOR_NAME, "registered_model_name", None
-        )
-        mlflow.pytorch.log_model(
-            pytorch_model=self.model,
-            artifact_path="model",
-            registered_model_name=registered_model_name,
-        )
-
-        if early_stop_callback is not None and self.checkpoint_callback.best_model_path:
-            mlflow.log_artifact(
-                local_path=self.checkpoint_callback.best_model_path,
-                artifact_path="restored_model_checkpoint",
-            )
-
-    client.flush(synchronous=True)
-
+    _flush_queue()
```

Comment:
```
    _flush_queue()
    return original(self, *args, **kwargs)
```

Does this work?

Reply:
mm why would you prefer this? if original() adds to the queue itself, wouldn't that just delay a flush?

Reply:
Does `original` add something to the queue?

Reply:
right. Makes sense.